### DISCUSSION TITLE
Does importMDX put the result into the compiled js files after yarn build?

### DISCUSSION THREAD

**Discussion by lefnire** [mdx-js/mdx#1363] [Q&A] [answered]
I'm using create-react-app, trying to understand how [importMDX](https://mdxjs.com/getting-started/create-react-app) works re: final build (via `yarn build`). I see `importMDX.sync('./file.mdx')` is synchronous, and my assumption is that means it gets compiled into the final files? Or is that not true, and files still get loaded via JavaScript on the client's browser? Sorry for my newbishness - trying to optimize compiled result and wrap my head.

**Comment by ChristianMurphy** [comment, marked as answer]
See https://github.com/mdx-js/mdx/discussions/1355#discussioncomment-148993 for how to use MDX with CRA.
the macro is in need of a full rewrite https://github.com/mdx-js/mdx/pull/589#issuecomment-499325405 there are a couple last pending items and the docs will be updated https://github.com/mdx-js/mdx/pull/589#issuecomment-602239520

**Comment by lefnire** [comment]
Thanks for the response!
